python/pyarrow/tests/test_serialization.py

```diff
@@ -285,6 +285,16 @@ def test_datetime_serialization(large_memory_map):
         for d in data:
             serialization_roundtrip(d, mmap)
 
+def test_torch_serialization(large_memory_map):
+    pytest.importorskip("torch")
+    import torch
+    with pa.memory_map(large_memory_map, mode="r+") as mmap:
+        # These are the only types that are supported for the
+        # PyTorch to NumPy conversion
+        for t in ["float32", "float64",
+                  "uint8", "int16", "int32", "int64"]:
+            obj = torch.from_numpy(np.random.randn(1000).astype(t))
+            serialization_roundtrip(obj, mmap)
```

Comment:
I wouldn't expect our `serialization_roundtrip` function to work on this. You may need to either modify the `assert_equal` function to handle torch tensors (don't forget to check both the values and the dtypes), or you may need to just explicitly compare the dtypes and the values.

Reply:
I fixed the tests (and made sure torch.equals only returns true if the datatypes also agree)